### PR TITLE
feat: delete connection records

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What?
- Implemented API to delete connection records by orgId
- Implemented store deleted audit trails functionality
- Error handling

### Why?
- To enable efficient management and removal of outdated or unnecessary connection data, ensuring data accuracy and system performance

### CODE REVIEW

File: apps/api-gateway/src/connection/connection.controller.ts [hunk 1]
```diff
@@ -21,7 +21,8 @@ import { IConnectionSearchCriteria } from '../interfaces/IConnectionSearch.inter
 import { SortFields } from 'apps/connection/src/enum/connection.enum';
 import { ClientProxy} from '@nestjs/microservices';
 import { QuestionAnswerWebhookDto, QuestionDto} from './dtos/question-answer.dto';
-
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (tipusinghaw) [inline apps/api-gateway/src/connection/connection.controller.ts:24]: Please remove `no-unused-vars`

File: apps/connection/src/connection.repository.ts [hunk 2]
```diff
@@ -316,4 +316,52 @@ export class ConnectionRepository {
       throw error;
     }
   }
+
+  async deleteConnectionRecordsByOrgId(orgId: string): Promise<IDeletedConnectionsRecord> {
+    const tablesToCheck = ['credentials', 'presentations'];
```
Comment (tipusinghaw) [inline apps/connection/src/connection.repository.ts:321]: Please create an enum for the table name, it may break when the table will migrated

Comment (bhavanakarwade) [reply apps/connection/src/connection.repository.ts:321]: Resolved

File: apps/api-gateway/src/connection/connection.controller.ts (hunk 1 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/connection/connection.controller.ts:24]: Resolved

**Pull request merged**